source/fundamentals/crud/restful-api-tutorial.txt

```diff
@@ -0,0 +1,377 @@
+
+      .. code-block:: json
+         :copyable: true
+
+         {
+            "Logging": {
+               "LogLevel": {
+               "Default": "Information",
+               "Microsoft.AspNetCore": "Warning"
+               }
+            },
+         "AllowedHosts": "*",
+         "MongoDB": {
+            "ConnectionURI": "<Atlas connection string>",
+            "DatabaseName": "sample_mflix",
+            "CollectionName": "playlist"
+            }
+         }
+
+      This tutorial uses the ``sample_mflix`` database and the ``playlist``
+      collection. Replace the ``<Atlas connection string>`` placeholder with
+      your MongoDB Atlas connection string. For more information on how to find
+      your connection string, see the :atlas:`Connect to Your Cluster </tutorial/connect-to-your-cluster>` tutorial.
+
+   .. step:: Create the service
+
+      In your project, create a folder named ``Services``. In the ``Services``
+      folder, create a new file named ``MongoDBService.cs`` and add the
+      following code:
+
+      .. literalinclude:: /includes/fundamentals/code-examples/restful-api-tutorial/MongoDBServiceSetup.cs
+         :language: csharp
+         :dedent:
+      
+      The preceding code defines a class named ``MongoDBService`` that includes
+      empty asynchronous functions. Throughout this tutorial, you add code to these
+      functions as you create your endpoints. The settings you configured in the
+      ``appsettings.json`` file are set to variables.
+
+   .. step:: Connect the service to the application
+
+      Open the ``Program.cs`` file and add the following code to the top of the file:
+
+      .. literalinclude:: /includes/fundamentals/code-examples/restful-api-tutorial/MongoDBExampleProgram.cs
+         :language: csharp
+         :dedent:
+         :start-after: start-program-setup
+         :end-before: end-program-setup
+
+      In the preceding code, the ``GetSection()`` function pulls your settings
+      from the ``MongoDB`` field in the ``appsettings.json`` file. 
+      
+      .. tip::
+         
+         If the template code already has the ``builder`` variable, don't add it twice. 
+
+   .. step:: Create the document model
+
+      Now that the service is set up, you can create a data model for your collection.
+
+      In the ``Models`` folder, create a new file named ``Playlist.cs`` and add
+      the following code:
+
+      .. literalinclude:: /includes/fundamentals/code-examples/restful-api-tutorial/PlaylistSetup.cs
+         :language: csharp
+         :dedent:
+
+      In the preceding code, the ``Id`` field is represented as an ``ObjectId``
+      in BSON and the ``_id`` fild within MongoDB. When you work with this
+      locally in your application, it is a string.
+
+      The ``movieIds`` field will be known as ``items``. When sending or
+      receiving JSON, the field will also be known as ``items`` instead of
+      ``movieIds``.
```

Comment:
S: adapt future tense
```suggestion
      The ``movieIds`` field  is serialized as ``items``. When you send or
      receive JSON values, the field is also named ``items`` instead of
      ``movieIds``.
```